File: backend/src/main/java/org/akashihi/mdg/entity/AccountType.java
package org.akashihi.mdg.entity;

public enum AccountType {
    INCOME,
    ASSET,
    EXPENSE;

    public String toDbValue() {
        return this.name().toLowerCase();
    }

    public static AccountType from(String status) {
        return AccountType.valueOf(status.toUpperCase());
    }
}

File: backend/src/main/java/org/akashihi/mdg/entity/AccountType.kt
package org.akashihi.mdg.entity

enum class AccountType {
    INCOME, ASSET, EXPENSE;

    fun toDbValue(): String {
        return name.lowercase()
    }

    companion object {
        @JvmStatic
        fun from(status: String): AccountType {
            return valueOf(status.uppercase())
        }
    }
}